### PR TITLE
feat(listing): adding renaming on keydown enter for OSX and open file for other OS

### LINKED ISSUE #1701: Handle "Enter" on file browser item
It should edit the file name on OSX and open the file on other platforms.

### PR DESCRIPTION
**Pull request by alexandercbooth**
Closes #1701.

Rename on "Enter" in OSX and opens file for other operating systems.

### CODE REVIEW

**Comment by alexandercbooth** [comment]
Out of curiosity, why do we want different behavior between the OSX and others?

File: src/filebrowser/listing.ts [hunk 1]
```diff
@@ -825,6 +825,38 @@ class DirListing extends Widget {
    */
   private _evtKeydown(event: KeyboardEvent): void {
     switch (event.keyCode) {
+    case 13: // Enter
+    // Do nothing if any modifier keys are pressed.
```
Comment (blink1073) [inline src/filebrowser/listing.ts]: Do you mind indenting this block one level?

File: src/filebrowser/listing.ts [hunk 2]
```diff
@@ -825,6 +825,38 @@ class DirListing extends Widget {
...
+    if (event.ctrlKey || event.shiftKey || event.altKey || event.metaKey) {
+      return;
+    }
+    event.preventDefault();
+    event.stopPropagation();
+
+    if (IS_MAC) {
+      this._doRename();
```
Comment (blink1073) [inline src/filebrowser/listing.ts]: we can return after this to avoid nesting the section below.

**Comment by blink1073** [comment]
Thanks again!  Just a copy of style comments.  

The reason for the difference is that Finder on OSX behaves that way, while Windows and Ubuntu both launch the file.

File: src/filebrowser/listing.ts (hunk 2 above)
Comment (alexandercbooth) [inline src/filebrowser/listing.ts]: Nice, that looks better, thanks!

**Comment by alexandercbooth** [comment]
My pleasure!

And that makes total sense to me, conforming to native experience feels the most natural. 
Thanks!

**Comment by blink1073** [comment]
Works like a charm on OSX and Windows, thanks!

**Comment by alexandercbooth** [comment]
Awesome! 😁

**Pull request merged**